packages/terra-functional-testing/src/reporters/file-output-reporter/index.js

```diff
@@ -0,0 +1,207 @@
+const fs = require('fs-extra');
+const path = require('path');
+const stripAnsi = require('strip-ansi');
+const SpecReporter = require('@wdio/spec-reporter').default;
+const endOfLine = require('os').EOL;
+const { Logger } = require('@cerner/terra-cli');
+
+const LOG_CONTEXT = '[terra-functional-testing:file-output-reporter]';
+
+class FileOutputReporter extends SpecReporter {
+  constructor() {
+    super();
+    this.runners = [];
+    this.resultJsonObject = {
+      startDate: '',
+      type: 'wdio',
+      locale: '',
+      formFactor: '',
+      theme: '',
+      output: {},
+      endDate: '',
+    };
+    this.fileName = '';
+    this.moduleName = process.cwd().split(path.sep).pop();
+
+    this.setResultsDir = this.setResultsDir.bind(this);
+    this.hasResultsDir = this.hasResultsDir.bind(this);
+    this.setTestModule = this.setTestModule.bind(this);
+    this.printReport = this.printReport.bind(this);
+    this.getMessage = this.getMessage.bind(this);
+
+    this.setResultsDir();
+    this.hasResultsDir();
+  }
+
+  /**
+   * Sets results directory for the test run. Uses the wdio reporterOptions.outputDir if set, otherwise
+   * it outputs to tests/wdio/reports.
+   * @return null;
+   */
+  setResultsDir() {
+    this.resultsDir = path.join(process.cwd(), 'tests', 'wdio', 'reports');
+  }
+
+  /**
+   * Check and create reports dir if doesn't exist
+   * @return null
+   */
+  hasResultsDir() {
+    if (!fs.existsSync(this.resultsDir)) {
+      fs.mkdirSync(this.resultsDir, { recursive: true }, (err) => {
+        if (err) {
+          Logger.error(err.message, { context: LOG_CONTEXT });
+        }
+      });
+    }
+  }
+
+  /**
+   * Formatting the filename based on locale, theme, and formFactor
+   * @return null
+   */
+  fileNameCheck({ formFactor, locale, theme }, browserName, cid) {
+    const fileNameConf = ['fileOutput'];
+    if (locale) {
+      fileNameConf.push(locale);
+      this.resultJsonObject.locale = locale;
+    }
+
+    if (theme) {
+      fileNameConf.push(theme);
+      this.resultJsonObject.theme = theme;
+    }
+
+    if (formFactor) {
+      fileNameConf.push(formFactor);
+      this.resultJsonObject.formFactor = formFactor;
+    }
+
+    if (browserName) {
+      fileNameConf.push(browserName);
+    }
+
+    if (cid) {
+      fileNameConf.push(cid);
+    }
+
+    this.fileName = fileNameConf.join('-');
+  }
+
+  /**
+   * Set the package name to moduleName property if specsValue contains /package string
+   * @param {string} specsValue - File path of current spec file from runners
+   * @return null
+   */
+  setTestModule(specsValue) {
+    const index = specsValue.lastIndexOf(`packages${path.sep}`);
```

Comment:
Should we also add `${path.sep}` in front of `packages` to be more robust? Similar to [this](https://github.com/cerner/terra-toolkit/blob/8dedba1178b2993bd751dc0c35704f36c0bbbd0c/packages/terra-functional-testing/src/reporters/spec-reporter/wdio-spec-reporter.js#L120)?